sim/core/rage.go

```diff
@@ -24,8 +28,19 @@ func (character *Character) EnableRageBar(startingRage float64) {
 
 				damage := hitEffect.Damage
 
-				// TODO: Put the actual equation here.
-				generatedRage := damage
+				if hitEffect.IsMH() {
+					HitFactor = 3.5 / 2
+					BaseSwingSpeed = character.AutoAttacks.MH.SwingSpeed
+				} else {
+					HitFactor = 1.75 / 2
+					BaseSwingSpeed = character.AutoAttacks.OH.SwingSpeed
+				}
+
+				if hitEffect.HitType == MeleeHitTypeCrit {
+					HitFactor *= 2
+				}
+
+				generatedRage := damage*RageFactor + HitFactor*BaseSwingSpeed
```

Comment:
It looks like we can delete the 'damage' variable above and inline hitEffect.Damage here.